### PR TITLE
Litespeed Kubernetes WebADC LoadBalancer submission for the DO market…

### PR DESCRIPTION
**Pull request by rperper**
…place

## BACKGROUND
* We would like to submit the LiteSpeed Kubernetes WebADC LoadBalancer for inclusion in the Digital Ocean Marketplace.

-----------------------------------------------------------------------

## Changes
* Changes as specified in the CONTRIBUTING.md file.

-----------------------------------------------------------------------

## Checklist
- [ ] review the [contributing doc](https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md) with steps for both adding or updating your application (if applicable)
------------------------------------------------------------------------

Reviewer: @marketplace-eng

### CODE REVIEW

**Comment by jgannondo** [comment]
@rperper sorry for the delay, we have been heads down on a major release and this slipped through the cracks. I've asked the team to take a look as soon as they can.

**Comment by bradford-hamilton** [comment]
Hey @rperper, when trying to run your new changes I am getting:

```
Error: failed to download "ls-k8s-webadc/ls-k8s-webadc" at version "0.1.13"
```

**Comment by rperper** [comment]
Hi @bradford-hamilton !  Sorry didn't think to look here!  Will do so more often.  We have a new version which should address that issue (0.1.15).  Please let me know if you run into anything else.
Thanks!  

Bob Perper
rperper@litespeedtech.com

**Comment by rperper** [comment]
@jgannondo it's been a bit since my last update.  Any problems getting to it?
Thanks,

Bob

**Comment by keladhruv** [comment]
Hi Roger, 

I had two suggestions for the PR -

1. Changing the folder name to something simpler, perhaps `litespeed-webadc`
2. I get the following error as a the installation files expect a license key -
```
"ls-k8s-webadc" has been added to your repositories
Creating trial license.  If you have a paid for license, you should update the secret using the commands:
   kubectl delete secret -n ls-k8s-webadc ls-k8s-webadc
   kubectl create secret generic -n ls-k8s-webadc ls-k8s-webadc --from-file=license=./license.key --from-file=serial=./serial.no
The connection to the server localhost:8080 was refused - did you specify the right host or port?
Error creating secret from trial key.  Fix error above and try again ```

**Comment by rperper** [comment]
Thanks for the feedback.  Let me look into it.  Should automatically get granted a trial license.
Thanks again, I'll be back to you shortly.

Bob

**Comment by rperper** [comment]
I created a fresh cluster and a fresh droplet to use as the installation client.  After putting kubectl and helm on the client I did the install.  It worked and it's running now.  It created the license as it should have.  

My experience is that the message `The connection to the server localhost:8080 was refused - did you specify the right host or port?` is from kubectl not being configured to speak the cluster (missing ~/.kube/config).  Is that something I need to do in my installation?  Helm calls kubectl directly, so it won't work unless kubectl is happy.

Since the folder name is in the marketplace/stacks folder we can change that easily enough.  The name of the kubernetes objects would be more difficult to change as they are coded in lots of places.  Is that ok?

Let me know what you think.  
Thanks!

Bob

**Comment by rperper** [comment]
@keladhruv just wondered if you'd had a chance to review my comments above?
Thanks,

Bob

File: stacks/ls-k8s-webadc/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+################################################################################
+# chart
+################################################################################
+STACK="ls-k8s-webadc"
+CHART="ls-k8s-webadc/ls-k8s-webadc"
+CHART_VERSION="0.1.15"
+NAMESPACE="ls-k8s-webadc"
+
+kubectl get secret ls-k8s-webadc -n ls-k8s-webadc > /dev/null 2>&1
+if [ "$?" -ne "0" ]; then
```
Comment (paulanunda) [inline stacks/ls-k8s-webadc/deploy.sh:20]: Could you simplify this logic a bit? Assume that the system installing it won't have a trail.key, serial.no, or license.key file.

**Review by paulanunda — CHANGES_REQUESTED** [review]
Thanks for the PR; I was able to get this to successfully install. Just 1 change requested.

**Comment by paulanunda** [comment]
@rperper to your questions:
>  Is that something I need to do in my installation? Helm calls kubectl directly, so it won't work unless kubectl is happy.
No. Our installation env will have kubectl in place.

> The name of the kubernetes objects would be more difficult to change as they are coded in lots of places. Is that ok?
The k8s object names are fine. `litespeed-webadc` as a folder name would be preferred.

File: stacks/ls-k8s-webadc/deploy.sh (hunk 1 above)
Comment (rperper) [reply stacks/ls-k8s-webadc/deploy.sh:20]: Removed the code which tested for a license and went on to create the trial license.

**Comment by rperper** [comment]
@paulanunda Directory renamed as requested, change to deploy made and pushed.
Thanks!

Bob

**Review by paulanunda — APPROVED** [review]

**Pull request merged**